browser_use/browser/browser.py

```diff
@@ -273,14 +291,40 @@ async def _setup_builtin_browser(self, playwright: Playwright) -> PlaywrightBrow
 			],
 		}
 
-		browser = await browser_class.launch(
-			headless=self.config.headless,
-			args=args[self.config.browser_class],
-			proxy=self.config.proxy,
-			handle_sigterm=False,
-			handle_sigint=False,
-		)
-		return browser
+		# Add profile directory to args if specified for Chromium
+		if self.config.profile_directory and self.config.browser_class == 'chromium':
+			args['chromium'].append(f'--profile-directory={self.config.profile_directory}')
+			logger.debug(f'Using profile directory: {self.config.profile_directory}')
+
+		# Base launch options for both methods
+		common_options = {
+			'headless': self.config.headless,
+			'args': args[self.config.browser_class] + self.disable_security_args + self.config.extra_browser_args,
```

Comment:
```suggestion
			'args': args[self.config.browser_class]
```

Those extra args are already added above